### PR TITLE
refactor(fs): Remove unused proxied functions

### PR DESCRIPTION
**Pull request by zharinov**
## Context

- Ref: #16125 

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [ ] Code inspection only, or
- [x] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

**Review by viceice — COMMENTED** [review]
wasn't the plan to change all FS io to our utils?

File: lib/util/fs/proxies.ts [hunk 1]
```diff
@@ -1,5 +1,5 @@
 import fs from 'fs-extra';
-import type { MoveOptions, WriteFileOptions } from 'fs-extra';
+import type { WriteFileOptions } from 'fs-extra';
 
 // istanbul ignore next
```
Comment (viceice) [inline lib/util/fs/proxies.ts:4]: shouldn't we add tests instead of remove the spec file completely?

**Review by viceice — DISMISSED** [review]

File: lib/util/fs/proxies.ts (hunk 1 above)
Comment (zharinov) [reply lib/util/fs/proxies.ts:4]: I'll test remaing files once they've refactored

File: lib/util/fs/index.spec.ts [hunk 2]
```diff
@@ -190,7 +190,7 @@ describe('util/fs/index', () => {
           const { dirFromEnv } = setupMock(tmpDir.path);
           const res = await ensureCacheDir('bundler');
           expect(res).toEqual(dirFromEnv);
-          expect(await exists(dirFromEnv)).toBeTrue();
+          expect(fs.existsSync(dirFromEnv)).toBeTrue();
```
Comment (viceice) [inline lib/util/fs/index.spec.ts:193]:
```suggestion
          expect(await fs.pathExists(dirFromEnv)).toBeTrue();
```

😉

**Review by viceice — COMMENTED** [review]
just a small thing

**Review by viceice — APPROVED** [review]

**Pull request merged**